radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/PeerManager.java

```diff
@@ -152,6 +156,61 @@ public PeerManager(
 		}
 	}
 
+	public CompletableFuture<PeerChannel> findOrCreateProxyChannel(NodeId nodeId) {
```

Comment:
By the way, why not `Promise`? It wraps uses `Result` as container and uses same `map/flatMap` notation. Overall this might help simplify code.

Reply:
Will look into, but I'd rather do that in a separate PR as this would require to modify the existing code for initiating the p2p connections (PeerOutboundBootstrap) which currently uses CompletableFutures.